erts/preloaded/src/erlang.erl

```diff
@@ -2612,7 +2614,7 @@ term_to_iovec(_Term, _Options) ->
 
 %% Shadowed by erl_bif_types: erlang:tl/1
 -spec tl(List) -> term() when
-      List :: [term(), ...].
+      List :: nonempty_maybe_improper_list().
```

Comment:
Note that how you write it in the code does not affect how it looks in the docs.

Reply:
[term(), ...] and nonempty_maybe_improper_list() are different types, right?

Reply:
Yes, didn't see that. But what you says makes the spec strange, considering we have success typings. The tail of a list() cannot be improper.

Reply:
I take back what I wrote. Considering the return value is term(), you're right. Hope I got it now.